eth/downloader/skeleton.go

```diff
@@ -437,7 +439,7 @@ func (s *skeleton) sync(head *types.Header) (*types.Header, error) {
 			// we don't seamlessly integrate reorgs to keep things simple. If the
 			// network starts doing many mini reorgs, it might be worthwhile handling
 			// a limited depth without an error.
-			if reorged := s.processNewHead(event.header, event.force); reorged {
+			if reorged := s.processNewHead(event.header, event.final, event.force); reorged {
```

Comment:
The final here is always nil, isn't it?

We have two ways to send head event:
- the event sent by `BeaconSync` in order to trigger the initial sync
- the event sent by `BeaconExtend` in order to integrate the newly arrived header

Finalized block will only be provided in `BeaconSync`, the corresponding event is not handled here, but here https://github.com/ethereum/go-ethereum/blob/master/eth/downloader/skeleton.go#L254

In summary, the finalized block will always be nil IIUC.